app/src/main/java/com/sunasterisk/getitdone/ui/home/HomePresenter.kt

```diff
@@ -0,0 +1,89 @@
+package com.sunasterisk.getitdone.ui.home
+
+import com.sunasterisk.getitdone.data.OnLoadedCallback
+import com.sunasterisk.getitdone.data.model.Task
+import com.sunasterisk.getitdone.data.model.TaskList
+import com.sunasterisk.getitdone.data.repository.TaskListRepository
+import com.sunasterisk.getitdone.data.repository.TaskRepository
+import com.sunasterisk.getitdone.data.source.local.TaskListLocalDataSource
+import com.sunasterisk.getitdone.data.source.local.TaskLocalDataSource
+import com.sunasterisk.getitdone.data.source.local.dao.TaskDAOImpl
+import com.sunasterisk.getitdone.data.source.local.dao.TaskListDAOImpl
+import com.sunasterisk.getitdone.data.source.local.database.AppDatabase
+import java.lang.Exception
+
+class HomePresenter private constructor() : HomeContract.Presenter {
+
+    private var view: HomeContract.View? = null
+
+    private lateinit var taskListRepository: TaskListRepository
+    private lateinit var taskRepository: TaskRepository
+
+    override fun attach(view: HomeContract.View) {
+        this.view = view
+
+        val database = AppDatabase.getInstance(view.getParentContext())
+
+        taskListRepository = TaskListRepository.getInstance(
+            TaskListLocalDataSource.getInstance(
+                TaskListDAOImpl.getInstance(database)
+            )
+        )
+
+        taskRepository = TaskRepository.getInstance(
+            TaskLocalDataSource.getInstance(
+                TaskDAOImpl.getInstance(database)
+            )
+        )
+    }
+
+    override fun detach() {
+        view = null
+    }
+
+    override fun getTaskListFromId(id: Int) {
+        taskListRepository.getTaskListFromId(id, object : OnLoadedCallback<TaskList> {
+            override fun onSuccess(data: TaskList) {
+                view?.setToolBarTitle(data.title)
+            }
+
+            override fun onFailure(exception: Exception) {
+                view?.handleException(exception)
+            }
+        })
+    }
+
+    override fun getTasksFromTaskListId(listId: Int) {
+        taskRepository.getAllTasksByListId(listId, object : OnLoadedCallback<List<Task>> {
+            override fun onSuccess(data: List<Task>) {
+                view?.handleLoadedTasks(data)
+            }
+
+            override fun onFailure(exception: Exception) {
+                view?.handleException(exception)
+            }
+        })
+    }
+
+    override fun updateTask(task: Task) {
+        taskRepository.updateTask(task, object : OnLoadedCallback<Boolean> {
+            override fun onSuccess(data: Boolean) {
+                if (data)
+                    view?.displayMessage("Update task successful")
```

Comment:
Thông báo hiển thị mà người dùng nhìn thấy cần được đặt trong strings.xml để tiện cho việc chuyển đổi ngôn ngữ 
Vì không có context nên mình có thể gửi resId lên 
```suggestion
                    view?.displayMessage(R.string.msg_update_task_successfully)
```